junit-platform-engine/src/main/java/org/junit/platform/engine/support/hierarchical/NodeTestTask.java

```diff
@@ -231,4 +245,20 @@ public void awaitFinished() throws InterruptedException {
 		}
 	}
 
+	private interface DynamicTaskState {
```

Comment:
```suggestion
	@FunctionalInterface
	private interface DynamicTaskState {
```

Reply:
Do you feel strongly about this? This is only used internally here and not user facing so we don't have to guarantee it will stay a SAM type (pun intended).

Reply:
Well, we use it as a functional interface at the moment. So I'm in favor of annotating it as such, for the sake of the reader/maintainers. 

Since it's not public, we could safely change our minds later if necessary.

Reply:
Done in fad47fd162ef00079c6abe9b19b7d78be41609b0.